executor/src/witgen/query_processor.rs

```diff
@@ -26,9 +28,11 @@ where
     pub fn process_queries_on_current_row(
         &mut self,
         rows: &RowPair<T>,
+        columns: &HashSet<PolyID>,
```

Comment:
I don't understand the need for a subset here. We could also just have a single column, that might be more logical.

Also note that we iterate on a hashset here, so the order is not deterministic.

Reply:
Sure, changed it so that it only a single column is processed. When this is called, columns are now iterated over in deterministic order and filtered that they need to be contained in the witness set (this is important, because otherwise a machine computes values for cells outside the machine).